### ISSUE TITLE
[Bug]: muti cn e2e bvt test case  does not return result in 60s and mo crash

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#7012]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.7.0 or 8b23a93):11be6e6d365d5b7cc2ac7669af6e05bdafb8789e
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

zz_accesscontrol/grant_privs_role.sql case reported MO does not return result in 60000 ms：

[ERROR]
[SCRIPT   FILE]: /home/runner/work/matrixone/matrixone/head/test/distributed/cases/zz_accesscontrol/grant_privs_role.sql
[ROW    NUMBER]: 214
[SQL STATEMENT]: create role 'role_account_priv_9';
[EXPECT RESULT]:
null
[ACTUAL RESULT]:
MO does not return result in 60000 ms.

mo crash：
![image](https://user-images.githubusercontent.com/113406637/204704718-ebcaa600-8af5-4fd9-8100-5134a83834c9.png)
part of log:
2022/11/28 13:32:36.223772 +0000 ERROR frontend/session.go:1445 GetTxn. error:txn write conflict 729d237e-6641-430d-b6cf-4b4b875b374f 72dacee6-2105-4993-a975-6e9542ee4466
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnHandler).GetTxn
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:1445
github.com/matrixorigin/matrixone/pkg/frontend.doUse
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1003
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).handleChangeDB
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1021
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3154
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3816
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:170
2022/11/28 13:32:28.895472 +0000 ERROR log-service log/logger.go:123 failed to refresh task storage {"uuid": "7c4dccb4-4d3c-41f8-b482-5251dc7a41bf", "error": "context deadline exceeded"}
2022/11/28 13:32:30.404953 +0000 ERROR morpc/message.go:41 error: invalid input: timeout has invalid deadline
github.com/matrixorigin/matrixone/pkg/common/morpc.RPCMessage.GetTimeoutFromContext
	/home/runner/work/matrixone/matrixone/head/pkg/common/morpc/message.go:41
github.com/matrixorigin/matrixone/pkg/common/morpc.(*server).startWriteLoop.func1
	/home/runner/work/matrixone/matrixone/head/pkg/common/morpc/server.go:267
github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1
	/home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:258
2022/11/28 13:32:30.400468 +0000 ERROR frontend/util.go:509 query trace status {"connection_id": 1805, "statement": "show tables", "status": "fail", "error": "illegal state", "span": {"trace_id": "0846baa2-6f21-11ed-ba91-000d3a633d4d", "kind": "statement"}, "session_info": "connectionId 1805"}
github.com/matrixorigin/matrixone/pkg/frontend.logError
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:509
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:493
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStatus
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/util.go:483
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3668
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:3816
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
	/home/runner/work/matrixone/matrixone/head/pkg/frontend/routine.go:170

**Comment by heni02** [comment]
[mo-service.log](https://github.com/matrixorigin/matrixone/files/10119337/mo-service.log)

**Comment by daviszhen** [comment]
we find the  `GetTxn. error:txn write conflict `

```

2022/11/28 13:32:36.223772 +0000 ERROR frontend/session.go:1445 GetTxn. error:txn write conflict 729d237e-6641-430d-b6cf-4b4b875b374f 72dacee6-2105-4993-a975-6e9542ee4466
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnHandler).GetTxn
/home/runner/work/matrixone/matrixone/head/pkg/frontend/session.go:1445
github.com/matrixorigin/matrixone/pkg/frontend.doUse
/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1003
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).handleChangeDB
/home/runner/work/matrixone/matrixone/head/pkg/frontend/mysql_cmd_executor.go:1021
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
```

**Comment by sukki37** [comment]
repro: https://github.com/matrixorigin/matrixone/actions/runs/3590152201

**Comment by XuPeng-SH** [comment]
the duplicate issue with #6971 
`dn` may send duplicate data to `cn` now and `cn` doesn't do an idempotence check. @aptend will add idempotence check in `cn`

**Comment by aptend** [comment]
fixed in 3e3dda573d8216282df061dbce237c49c28bdfa7. @heni02 pls check

**Comment by heni02** [comment]
confirm，closed
https://github.com/matrixorigin/matrixone/actions/runs/3590017333/jobs/6043002910
